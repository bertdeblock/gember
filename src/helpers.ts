import type { EmberPackageJson } from "./types.js";

export function isV1Addon(packageJson: EmberPackageJson): boolean {
  if (isAddon(packageJson)) {
    const { version } = packageJson["ember-addon"] ?? {};

    return version === 1 || version === undefined;
  }

  return false;
}

export function isV2Addon(packageJson: EmberPackageJson): boolean {
  if (isAddon(packageJson)) {
    const { version } = packageJson["ember-addon"] ?? {};

    return version === 2;
  }

  return false;
}

function isAddon(packageJson: EmberPackageJson): boolean {
  if (Array.isArray(packageJson.keywords)) {
    return packageJson.keywords.includes("ember-addon");
  }

  return false;
}
