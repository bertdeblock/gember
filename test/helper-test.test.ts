import { afterEach, it } from "vitest";
import { Package } from "./helpers.ts";

let pkg: Package;

afterEach(() => pkg.cleanUp());

it("generates a `.gjs` helper-test", async (ctx) => {
  pkg = await Package.create("v2-addon");

  await pkg.gember("helper-test", "foo");

  const content = await pkg.readFile("tests/integration/helpers/foo-test.gjs");

  ctx.expect(content).toMatchSnapshot();
});

it("generates a `.gjs` helper-test at a custom path", async (ctx) => {
  pkg = await Package.create("v2-addon");

  await pkg.gember("helper-test", "foo", "--path=tests/foo");

  const content = await pkg.readFile("tests/foo/foo-test.gjs");

  ctx.expect(content).toMatchSnapshot();
});

it("generates a `.gts` helper-test", async (ctx) => {
  pkg = await Package.create("v2-addon");

  await pkg.gember("helper-test", "foo", "--ts");

  const content = await pkg.readFile("tests/integration/helpers/foo-test.gts");

  ctx.expect(content).toMatchSnapshot();
});

it("generates a `.gts` helper-test at a custom path", async (ctx) => {
  pkg = await Package.create("v2-addon");

  await pkg.gember("helper-test", "foo", "--path=tests/foo", "--ts");

  const content = await pkg.readFile("tests/foo/foo-test.gts");

  ctx.expect(content).toMatchSnapshot();
});
