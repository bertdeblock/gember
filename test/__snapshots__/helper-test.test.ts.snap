// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`generates a \`.gjs\` helper-test 1`] = `
"import { render } from '@ember/test-helpers';
import { module, test } from 'qunit';
import { setupRenderingTest } from 'v2-addon/tests/helpers';

module('Integration | Helper | foo', function (hooks) {
  setupRenderingTest(hooks);

  test('it renders', async function (assert) {
    const inputValue = '1234';

    await render(<template>{{foo inputValue}}</template>);

    assert.dom().hasText('1234');
  });
});
"
`;

exports[`generates a \`.gjs\` helper-test at a custom path 1`] = `
"import { render } from '@ember/test-helpers';
import { module, test } from 'qunit';
import { setupRenderingTest } from 'v2-addon/tests/helpers';

module('Integration | Helper | foo', function (hooks) {
  setupRenderingTest(hooks);

  test('it renders', async function (assert) {
    const inputValue = '1234';

    await render(<template>{{foo inputValue}}</template>);

    assert.dom().hasText('1234');
  });
});
"
`;

exports[`generates a \`.gts\` helper-test 1`] = `
"import { render } from '@ember/test-helpers';
import { module, test } from 'qunit';
import { setupRenderingTest } from 'v2-addon/tests/helpers';

module('Integration | Helper | foo', function (hooks) {
  setupRenderingTest(hooks);

  test('it renders', async function (assert) {
    const inputValue = '1234';

    await render(<template>{{foo inputValue}}</template>);

    assert.dom().hasText('1234');
  });
});
"
`;

exports[`generates a \`.gts\` helper-test at a custom path 1`] = `
"import { render } from '@ember/test-helpers';
import { module, test } from 'qunit';
import { setupRenderingTest } from 'v2-addon/tests/helpers';

module('Integration | Helper | foo', function (hooks) {
  setupRenderingTest(hooks);

  test('it renders', async function (assert) {
    const inputValue = '1234';

    await render(<template>{{foo inputValue}}</template>);

    assert.dom().hasText('1234');
  });
});
"
`;
